components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/UUFServer.java

```diff
@@ -171,6 +173,18 @@ public void serve(HttpRequest request, HttpResponse response) {
             requestDispatcher.serveDefaultErrorPage(STATUS_INTERNAL_SERVER_ERROR, msg, response);
         }
 
+        // Logic to validate against CSRF attacks
+        if (request.getMethod().equals("POST") &&
+                !app.getConfiguration().getCsrfIgnoreUris().contains(request.getUriWithoutContextPath())) {
```

Comment:
Instead of string comparison, we need to URI pattern matching. We can use `UriPattern` class for this.